src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -1205,41 +1234,81 @@ private boolean isSuspended()
+             * frame were either:
+             * a) an old frame of an adaptive stream OR
+             * b) a frame from an adaptive stream on a layer we're not using
+             * @return true if we're accepting this frame
+             */
+            private boolean isAccepting()
+            {
+                return srcSeqNumLimit != -1;
+            }
+
+            /**
+             * Update this {@link SeenFrame} with information from its
+             * corresponding {@link FrameDesc}
+             * @param frame the {@link FrameDesc} from which to update
+             */
+            void update(FrameDesc frame)
+            {
+                if (this == mostRecentSentFrame /* the max frame can expand */
+                    || availableQualityIndices == null
+                    || availableQualityIndices.length < 2)
+                {
+                    if (frame.lastSequenceNumberKnown())
+                    {
+                        srcSeqNumLimit = frame.getEnd();
+                    }
+                    else
+                    {
+                        srcSeqNumLimit = frame.getMaxSeen();
+                    }
+                }
+                if (isAccepting())
+                {
+                    if (frame.firstSequenceNumberKnown())
+                    {
+                        srcSeqNumStart = frame.getStart();
+                    }
+                    else
+                    {
+                        srcSeqNumStart = frame.getMinSeen();
+                    }
+                }
+            }
+
+            /**
              * Defines an RTP packet filter that controls which packets to be
              * written into some arbitrary target/receiver that owns this
-             * {@link SeenFrame}.
+             * {@link SeenFrame}.  {@link SeenFrame#update(FrameDesc)} should
+             * be called with the {@link FrameDesc} for this packet before
+             * calling accept.
              *
-             * @param source the {@link FrameDesc} that the RTP packet belongs
-             * to.
              * @param pkt the packet for which to decide to accept
              * @return <tt>true</tt> to allow the specified packet/<tt>buffer</tt>
              * to be written into the arbitrary target/receiver that owns this
              * {@link SeenFrame} ; otherwise, <tt>false</tt>
              */
-            boolean accept(FrameDesc source, RawPacket pkt)
+            boolean accept(RawPacket pkt)
             {
-                if (this == mostRecentSentFrame /* the max frame can expand */
-                    || availableQualityIndices == null
-                    || availableQualityIndices.length < 2)
-                {
-                    int end = source.getEnd();
-                    srcSeqNumLimit
-                        = end != -1 ? source.getEnd() : source.getMaxSeen() ;
-                }
-
-                if (srcSeqNumLimit == -1)
+                if (!isAccepting())
```

Comment:
You mentioned that you might want to change isAccepting and I agree. I don't think it provides enough information on what accepting means. Maybe a phrase that describes why it is or is not accepting is more appropriate. Normally I'd say it's good enough but you are focusing on renaming so...